src/main/java/speakql/SpeakQLSpeechlet.java

```diff
@@ -43,375 +34,410 @@
-     */
-    private SpeechletResponse handleNextEventRequest(Session session) {
-        String cardTitle = "More events on this day in history";
-        ArrayList<String> events = (ArrayList<String>) session.getAttribute(SESSION_TEXT);
-        int index = (int) session.getAttribute(SESSION_INDEX);
-        String speechOutput = "";
-        String cardOutput = "";
-        if (events == null) {
-            speechOutput =
-                    "With History Buff, you can get historical events for any day of the year."
-                            + " For example, you could say today, or August thirtieth."
-                            + " Now, which day do you want?";
-        } else if (index >= events.size()) {
-            speechOutput =
-                    "There are no more events for this date. Try another date by saying, "
-                            + " get events for august thirtieth.";
-        } else {
-            StringBuilder speechOutputBuilder = new StringBuilder();
-            StringBuilder cardOutputBuilder = new StringBuilder();
-            for (int i = 0; i < PAGINATION_SIZE && index < events.size(); i++) {
-                speechOutputBuilder.append("<p>");
-                speechOutputBuilder.append(events.get(index));
-                speechOutputBuilder.append("</p> ");
-                cardOutputBuilder.append(events.get(index));
-                cardOutputBuilder.append(" ");
-                index++;
-            }
-            if (index < events.size()) {
-                speechOutputBuilder.append(" Wanna go deeper in history?");
-                cardOutputBuilder.append(" Wanna go deeper in history?");
-            }
-            session.setAttribute(SESSION_INDEX, index);
-            speechOutput = speechOutputBuilder.toString();
-            cardOutput = cardOutputBuilder.toString();
-        }
-        String repromptText = "Do you want to know more about what happened on this date?";
-
-        // Create the Simple card content.
-        SimpleCard card = new SimpleCard();
-        card.setTitle(cardTitle);
-        card.setContent(cardOutput.toString());
-
-        SpeechletResponse response = newAskResponse("<speak>" + speechOutput + "</speak>", "<speak>" + repromptText + "</speak>");
-        response.setCard(card);
-        return response;
+    return parseJson(text);
+  }
+
+  /**
+   * Parse JSON-formatted list of events/births/deaths from Wikipedia, extract
+   * list of events and split the events into a String array of individual
+   * events. Run Regex matchers to make the list pretty by adding a comma after
+   * the year to add a pause, and by removing a unicode char.
+   * 
+   * @param text The JSON formatted list of events/births/deaths for a certain
+   *     date.
+   * @return String array of events for that date, 1 event per element of the
+   *     array.
+   */
+  private ArrayList<String> parseJson(String text) {
```

Comment:
I didn't prune the logic here yet, I assumed that you kept HistoryBuff stuff for a reason